### PR TITLE
Generate MTL and MPS structs and enums with Clang.jl

### PR DESCRIPTION
**Pull request by christiangnrd**
Currently works for structs and enums in Metal.framework with a few hacks and this branch of Clang devved: https://github.com/christiangnrd/Clang.jl/tree/objectiveC

Any advice, suggestions, or help welcome!

### CODE REVIEW

**Comment by maleadt** [comment]
Even if only for enums etc, this would be pretty useful! I imagined having to write our own Clang.jl-based generator in order to generate ObjC calls, but this would probably be a good first step.

**Comment by christiangnrd** [comment]
> I imagined having to write our own Clang.jl-based generator in order to generate ObjC calls, but this would probably be a good first step.

I think most of it can be upstreamed to Clang.jl. The changes I've done on my branch are either bug fixes or definitions of the `Generators` types for Objective-C constructs.

**Comment by maleadt** [comment]
Please ping when this is ready for review; you've marked it as such, but more commits keep appearing 🙂 

Also, were the necessary changes to Clang.jl merged?

**Comment by christiangnrd** [comment]
> Please ping when this is ready for review; you've marked it as such, but more commits keep appearing 🙂

I do have a terrible habit of refining my PRs after marking ready. I'll be more mindful of that in the future.

> Also, were the necessary changes to Clang.jl merged?

Not yet, https://github.com/JuliaInterop/Clang.jl/pull/519 and https://github.com/JuliaInterop/Clang.jl/pull/522 are general improvements/bug fixes to handle the types of enums that Apple ObjectiveC headers love so much (elaborated and attributes). I just created https://github.com/JuliaInterop/Clang.jl/pull/524 which once the other 2 are merged, shouldn't be too hard to review.

I just rebased and any further changes other than review changes will be for a future PR.

File: res/wrap/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# res/wrap/
```
Comment (maleadt) [inline res/wrap/README.md:1]:
```suggestion
# Automated wrapper generation
```

**Review by maleadt — APPROVED** [review]
LGTM. We could probably also remove some enums from `compiler/library.jl`, but that can happen in a different PR.

**Pull request merged**